### PR TITLE
[Storage] Migrate New-AzStorageQueueSASToken to Track2 SDK 

### PR DESCRIPTION
**Pull request by yifanz7**
## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Storage/Storage/Common/SasTokenHelper.cs [hunk 1]
```diff
@@ -328,7 +345,115 @@ public static ShareSignedIdentifier GetShareSignedIdentifier(ShareClient share,
...
+                    if (protocol.Value == SharedAccessProtocol.HttpsOrHttp)
+                    {
+                        sasBuilder.Protocol = SasProtocol.HttpsAndHttp;
+                    }
+                    else //HttpsOnly
+                    {
+                        sasBuilder.Protocol = SasProtocol.Https;
+                    }
+                }
+            }
```
Comment (blueww) [inline src/Storage/Storage/Common/SasTokenHelper.cs:451]: iPAddressOrRange , protocol  should be move 1 level out.
No matter use policy or not, we should always check iPAddressOrRange , protocol .

File: src/Storage/Storage/Common/Util.cs [hunk 2]
```diff
@@ -797,6 +798,25 @@ public static ShareClient GetTrack2ShareReference(string shareName, AzureStorage
...
+
+            string connectionString = context.ConnectionString;
+            if (context != null && context.StorageAccount != null && context.StorageAccount.Credentials != null && context.StorageAccount.Credentials.IsSAS)
+            {
+                connectionString = connectionString.Replace("SharedAccessSignature=?", "SharedAccessSignature=");
+            }
+
+            QueueClient queueClient;
+
+            queueClient = new QueueClient(connectionString, queueName);
```
Comment (blueww) [inline src/Storage/Storage/Common/Util.cs:816]: should also add clientoption to add useragent.

File: src/Storage/Storage/Common/AzureStorageQueue.cs [hunk 3]
```diff
@@ -98,7 +98,7 @@ public AzureStorageQueue(CloudQueue queue)
         }
 
         // Convert Track1 queue object to Track 2 queue Client
-        protected static QueueClient GetTrack2QueueClient(CloudQueue cloudQueue, AzureStorageContext context)
+        public static QueueClient GetTrack2QueueClient(CloudQueue cloudQueue, AzureStorageContext context)
```
Comment (blueww) [inline src/Storage/Storage/Common/AzureStorageQueue.cs:101]: It looks this function is not used out side of [AzureStorageQueue.cs](https://github.com/Azure/azure-powershell/pull/21966/files#diff-8ec26b2a5e1a42c0015e613ca99d811ced1d8911e7d998b51256bd6bbc36b99f)

Comment (yifanz7) [reply src/Storage/Storage/Common/AzureStorageQueue.cs]: Thanks for catching. I changed this for experiments and forgot to change it back, but this is not related to this cmdlet or useful. 
Reverted it back. 

RE: adding clientoptions to the function. Yes I'll add it in a later PR.

**Review by blueww — APPROVED** [review]

**Review by blueww — APPROVED** [review]

**Review by vidai-msft — APPROVED** [review]

**Pull request merged**